modules/management-console-agent/src/main/java/org/apache/ignite/internal/agent/action/controller/QueryActionsController.java

```diff
@@ -89,6 +89,27 @@ public QueryActionsController(GridKernalContext ctx) {
         return new IgniteFinishedFutureImpl<>();
     }
 
+    /**
+     * @param globalQryId Global query id.
+     * @return Kill query result.
+     */
+    public IgniteInternalFuture<QueryResult> kill(String globalQryId) {
+        return ctx.closure().callLocalSafe(() -> {
+            SqlFieldsQuery qryFields = new SqlFieldsQuery("KILL QUERY '" + globalQryId + "';");
```

Comment:
Can we receive SQL injections here?

Reply:
I don't think we should care too much about this here. If the user has access to killing queries, she probably can execute them as well.

Reply:
We have a special action for execution any SQL query, :)